packages/climate-ref/tests/unit/executor/test_hpc_executor.py

```diff
@@ -74,12 +74,28 @@ def test_join(self, metric_definition, tmp_path):
 
         assert len(executor.parsl_results) == 0
 
-    def test_join_exception(self, metric_definition, tmp_path):
+    def test_join_diagnostic_exception(self, metric_definition, tmp_path):
+        executor = HPCExecutor(log_dir=tmp_path / "parsl_runinfo")
+        future = futures.AppFuture(1)
+        executor.parsl_results = [ExecutionFuture(future, definition=metric_definition, execution_id=None)]
+
+        execution_result = ExecutionResult(
+            definition=metric_definition,
+            successful=False,
+            output_bundle_filename=None,
+            metric_bundle_filename=None,
+        )
+
+        future.set_exception(DiagnosticError("Some thing bad went wrong", execution_result))
+
+        executor.join(0.1)
```

Comment:
Is there anything that you can assert about after joining?

Reply:
Good point. I will add two assertions: (1) the result in the error is the same as the execution result, (2) the parsl_results shall be empty after join.